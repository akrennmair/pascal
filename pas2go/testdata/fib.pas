program fib;

var max : integer;
	fib1, fib2 : integer;
	nextNum : integer;


begin
	write('generating fibonacci numbers to what maximum? ');
	readln(max);

	fib1 := 1;
	fib2 := 1;

	writeln(fib1);

	while fib2 < max do
	begin
		writeln(fib2);
		nextNum := fib1 + fib2;
		fib1 := fib2;
		fib2 := nextNum
	end
end.
