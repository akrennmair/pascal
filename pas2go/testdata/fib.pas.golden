package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program fib
func main() {
	var (
		max     int
		fib1    int
		fib2    int
		nextnum int
	)
	_ = max
	_ = fib1
	_ = fib2
	_ = nextnum

	system.Write("generating fibonacci numbers to what maximum? ")
	system.Readln(&max)
	fib1 = 1
	fib2 = 1
	system.Writeln(fib1)
	for fib2 < max {

		system.Writeln(fib2)
		nextnum = fib1 + fib2
		fib1 = fib2
		fib2 = nextnum
	}
}
